src/new/wallet.js

```diff
@@ -765,44 +765,128 @@ class HathorWallet extends EventEmitter {
    * Send a transaction from its outputs
    * Currently does not have support to send custom tokens, only HTR
    *
-   * @param {Array} outputs Array of outputs with each element as an object with {'address', 'value'}
-   * @param {Array} inputs Array of inputs with each element as an object with {'hash', 'index'}
+   * @param {Array} outputs Array of outputs with each element as an object with {'address', 'value', 'timelock', 'token'}
+   * @param {Array} inputs Array of inputs with each element as an object with {'hash', 'index', 'token'}
    * @param {Object} token Token object {'uid', 'name', 'symbol'}. Optional parameter if user already set on the class
    * @param {Object} options Options parameters
    *  {
    *   'changeAddress': address of the change output
+   *   'startMiningTx': boolean to trigger start mining (default true)
    *  }
    *
    * @return {Promise} Promise that resolves when transaction is sent
    **/
-  sendManyOutputsTransaction(outputs, inputs = [], token = null, options = { changeAddress: null }) {
-    // XXX To accept here multi tokens in the same tx would be a bit more complicated because
-    // the method prepareSendTokensData would need a bigger refactor.
-    // I believe we should refactor all of that code (and it will be done on wallet-service)
-    // so I decided to change this method to accept only one token, which is enough for our needs right now
+  sendManyOutputsTransaction(outputs, inputs = [], token = null, options = {}) {
     storage.setStore(this.store);
+    const newOptions = Object.assign({ changeAddress: null, startMiningTx: true }, options);
     const txToken = token || this.token;
-    const isHathorToken = txToken.uid === HATHOR_TOKEN_CONFIG.uid;
-    const data = {
-      tokens: isHathorToken ? [] : [txToken.uid],
-      inputs: [],
-      outputs: [],
-    };
+    const tokensData = {};
+    const tokens = [];
+
+    // PrepareSendTokensData method expects all inputs/outputs for each token
+    // then the first step is to separate the inputs/outputs for each token
+    const defaultData = { outputs: [], inputs: [] };
+    if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
+      tokensData[txToken.uid] = Object.assign({}, defaultData);
+      tokens.push(txToken.uid);
+    } else {
+      for (const output of outputs) {
+        if (output.token && output.token !== HATHOR_TOKEN_CONFIG.uid) {
+          if (!(output.token in tokensData)) {
+            tokens.push(output.token);
```

Comment:
Isn't it better to remove the variable `tokens` and, after this loop, generate it from the keys of `tokensData`?